### PR TITLE
feat: added user service implementations for no-ldap profile

### PR DESCRIPTION
**Pull request by lmoesle**
---
name: Pull request
about: Contribute your changes to the project
title: '[pull-request]'
labels: ''
assignees: ''
---
**Description**
Short description or comments

**Reference**
Issues #XXX

**Checklist**

- [ ] Acceptance criteria are met
- [ ] Pipeline has been run successfully
- [ ] JUnit tests have been run successfully
- [ ] Changelog is adjusted
- [ ] [Documentations](https://git.muenchen.de/groups/digitalisierung/-/wikis/Dokumentationen) are completed
- [ ] Frontend is tested
- [ ] Created sub-branches are deleted
- [ ] Boards are updated ( [Digitalisierung](https://git.muenchen.de/groups/digitalisierung/-/boards) , [Support](https://git.muenchen.de/digitalisierung/digiwf-support) , [Zenhub](https://app.zenhub.com/workspaces/digiwf-621f70bf50ea1100120b7e93/board) )
- [ ] feature process is created or extended
- [ ] [git-ops](https://git.muenchen.de/digitalisierung/digiwf-ops) is customized
- [ ] Openshift environments are prepared (Secrets, etc.)

### CODE REVIEW

File: digiwf-engine/digiwf-engine-service/src/main/java/io/muenchendigital/digiwf/legacy/user/domain/service/UserServiceLdap.java [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+/**
+ * Service to query for users.
+ *
+ * @author externer.dl.horn
+ */
+@Profile("!no-ldap")
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class UserServiceLdap implements UserService {
```
Comment (dominikhorn93) [inline digiwf-engine/digiwf-engine-service/src/main/java/io/muenchendigital/digiwf/legacy/user/domain/service/UserServiceLdap.java]: LdapUserServiceImpl

File: digiwf-engine/digiwf-engine-service/src/main/java/io/muenchendigital/digiwf/legacy/user/domain/service/UserServiceNoLdap.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+/**
+ * Fake implementation of the UserService to enable local development without an ldap.
+ *
+ * @author ext.dl.moesle
+ */
+@Profile("no-ldap")
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class UserServiceNoLdap implements UserService {
```
Comment (dominikhorn93) [inline digiwf-engine/digiwf-engine-service/src/main/java/io/muenchendigital/digiwf/legacy/user/domain/service/UserServiceNoLdap.java]: UserServiceMock?

**Review by dominikhorn93 — CHANGES_REQUESTED** [review]
2 naming questions... but not sure about it

File: digiwf-engine/digiwf-engine-service/src/main/java/io/muenchendigital/digiwf/legacy/user/domain/service/UserServiceNoLdap.java (hunk 2 above)
Comment (lmoesle) [reply digiwf-engine/digiwf-engine-service/src/main/java/io/muenchendigital/digiwf/legacy/user/domain/service/UserServiceNoLdap.java]: `MockUserServiceImpl`? Dann ist es dasselbe Pattern wie auch beim `LdapUserServiceImpl?`

**Review by dominikhorn93 — APPROVED** [review]

**Pull request merged**